docs/standard/serialization/system-text-json/custom-contracts.md

```diff
@@ -0,0 +1,120 @@
+| Customize number handling for a specific type | `JsonTypeInfoKind.None` | Modify the <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo.NumberHandling?displayProperty=nameWithType> value for the type. | [Allow int values to be strings](#example-allow-int-values-to-be-strings) |
+
+## Example: Increment a property's value
+
+Consider the following example where the modifier increments the value of a certain property on deserialization by modifying its <xref:System.Text.Json.Serialization.Metadata.JsonPropertyInfo.Set?displayProperty=nameWithType> delegate. Besides defining the modifier, the example also introduces a new attribute that it uses to locate the property whose value should be incremented. This is an example of *customizing a property*.
+
+:::code language="csharp" source="snippets/custom-contracts/SerializationCount.cs":::
+
+Notice in the output that the value of `RoundTrips` is incremented each time the `Product` instance is deserialized.
+
+## Example: Serialize private fields
+
+By default, `System.Text.Json` ignores private fields and properties. This example adds a new class-wide attribute, `JsonIncludePrivateFieldsAttribute`, to change that default. If the modifier finds the attribute on a type, it adds all the private fields on the type as new properties to <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo>.
+
+:::code language="csharp" source="snippets/custom-contracts/PrivateFields.cs":::
+
+> [!TIP]
+> If your private field names start with underscores, consider adding a JSON naming policy to remove the underscore and capitalize the name.
+
+## Example: Ignore properties with a specific type
+
+Perhaps your model has properties with specific names or types that you don't want to expose to users. For example, you might have a property that stores credentials or some information that's useless to have in the payload.
+
+The following example shows how to filter out properties with a specific type, `SecretHolder`. It does this by first clearing the <xref:System.Text.Json.Serialization.Metadata.JsonTypeInfo.Properties?displayProperty=nameWithType> list and then readding only those properties that don't have the specified type. The filtered properties will completely disappear from the contract, which means `System.Text.Json` won't look at them either during serialization or deserialization.
+
+:::code language="csharp" source="snippets/custom-contracts/IgnoreType.cs":::
+
+## Example: Allow int values to be strings
+
+Perhaps your input JSON can contain quotes around one of the numeric types but not on others. If you had control over the class, you could place <xref:System.Text.Json.Serialization.JsonNumberHandlingAttribute> on the type to fix this, but you don't. Before .NET 7, you'd need to write a [custom converter](converters-how-to.md) to fix this behavior, which is hard. Using contract customization, you can customize the number handling behavior for any type.
```

Comment:
Custom converters aren't necessarily hard, but they certainly involve a degree of boilerplate not proportional to what needs to be achieved in this context
```suggestion
Perhaps your input JSON can contain quotes around one of the numeric types but not on others. If you had control over the class, you could place <xref:System.Text.Json.Serialization.JsonNumberHandlingAttribute> on the type to fix this, but you don't. Before .NET 7, you'd need to write a [custom converter](converters-how-to.md) to fix this behavior, which is involved. Using contract customization, you can customize the number handling behavior for any type.
```